packages/salesforcedx-vscode-core/test/vscode-integration/commands/functions/forceFunctionContainerStartCommand.test.ts

```diff
@@ -0,0 +1,102 @@
+import { vscodeStub } from '@salesforce/salesforcedx-utils-vscode/test/unit/commands/mocks';
+import * as proxyquire from 'proxyquire';
+import { assert, createSandbox, SinonSandbox, SinonStub, stub } from 'sinon';
+
+const proxyquireStrict = proxyquire.noCallThru();
+
+describe.only('Force Function Start Container Command Unit Tests.', () => {
+  let sandbox: SinonSandbox;
+
+  before(() => {
+    sandbox = createSandbox();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('Verify container start command.', () => {
+    const fakeUri: any = 'what';
+    let validateStartFunctionsUriStub: SinonStub;
+    let commandletRunStub: SinonStub;
+    let forceFunctionContainerStartCommand: any;
+    let FUNCTION_CONTAINER_LOG_NAME: string;
+    let CONTAINER_START_TEXT_KEY: string;
+    let filePathGathererConstructorStub: SinonStub;
+    let sfdxWorkspaceCheckerConstructorStub: SinonStub;
+    let forceFunctionContainerStartExecutorConstructorStub: SinonStub;
+
+    beforeEach(() => {
```

Comment:
So I probably spent entirely too much time trying to figure out how to deal with the `vscode` module when developing unit tests locally.  The issue is it doesn't exist locally so you can't mock it (or even run a test that imports code where it gets referenced) when running the tests outside of vscode (the ide). 

I found a work around that I don't love, but found it to be the lesser of the two evils of not writing unit tests.
Using proxyquire I can ensure we are stubbing all dependancies.  The downside is I'm losing the type checking when calling the function under test (note this impact the typescript compile for the test only).